### ISSUE TITLE
investigate fixes for macOS Big Sur installer if `apfs.util -B` is indeed removed

### ISSUE THREAD

**Issue by abathur** [NixOS/nix#3957]
I want to go ahead and put a pin in a [Big Sur installation issue I saw on Discourse](https://discourse.nixos.org/t/installing-on-macos-11-big-sur/8739):

> I can’t get the firmlinks thing working. Because (1) apfs.util doesn’t have the `-B` anymore, which I resolved by just doing a reboot.

IIRC, someone discovered `apfs.util -B` (`apfs.util --help` describes the purpose of this option as "bootstrap the root filesystem") as a way to skirt what had previously appeared to be a required reboot between adding /nix to synthetic.conf and being able to actually make the /nix directory. Here it is in the installer:

https://github.com/NixOS/nix/blob/c129e7c8f4846fcc146cfb3e408a75dd73794794/scripts/create-darwin-volume.sh#L135

If this is indeed true and the functionality hasn't moved elsewhere (or isn't a bug; or isn't something we can get restored by griping), I think our install process is going to necessarily become a clunky two-phase thing?

**Comment by abathur** [comment]
I was going to try to investigate this on a spare laptop, but after enrolling it in the beta program and upgrading it, I'm having trouble getting it through a full clean install, and I probably won't have time to figure that out for a bit 😬

**Comment by abathur** [comment]
I finally got this system back up and running on a clean install of Big Sur. I notice that apfs.util has a few new new options, including:

`-t stitches and creates synthetic objects on root volume group`

I played with this a little, and at a blush it appears to do what the `-B` flag did. If I add a name to /etc/synthetic.config, then run `apfs.util -t`, it does create the directory.

For reference, here are the other new flags I see:
```
-X set=yes|no path : set/unset embedded xattr max size enforcement on volume specified by 'path'
-z path : make the file at path exempt from copy-on-write semantics for snapshots'
-E path : drop the extents for 'path' from the first snapshot
```

I also noticed that the flag `-y path : get the sync-root flag from the first parent that has one for the item identified by 'path'`, which appears in the help output on Catalina but does not work, is functional in Big Sur.
